docs/proposals/20200506-conditions.md

```diff
@@ -0,0 +1,363 @@
+
+## Motivation
+
+- Cluster API requires a large number of objects to define a workload cluster: Cluster,
+  InfrastructureCluster, KubeadmControlPlane, MachineDeployments, Machines, and so on.
+  Each object carries its own status. Operators need to describe each object individually to
+  determine an overall cluster health summary.
+- Objects can be tied to each other directly (via object references), or indirectly
+  (via owner references). These dependencies can often require a deep knowledge of the
+  project’s internals and its APIs to investigate issues or problems. 
+- Some objects, like MachineSets or MachineDeployments, are considered a collection of
+  other objects (e.g. MachineSets). Their status is effectively tied to the status of each
+  item in the collection. 
+
+### Goals
+
+In order to make this proposal actionable in the v0.3.x timeframe, the proposal is going to
+be developed in an incremental fashion in three iterations. 
+
+The goal of this first iteration is:
+
+1. To agree on a design for the condition types.
+2. To provide backward-compatible support for `Phase`, `ErrorMessage`, `ErrorReason` fields.
+3. To provide utilities and libraries for a unified way to work with conditions.
+4. To provide flexibility for providers to incrementally adopt this proposal; or add, remove,
+   or update their own set of conditions.
+
+Once there will be an agreed design for the condition types, a second iteration will start
+with the following goals:
+
+1. To define the first set of conditions on the different Cluster API objects.
+2. To have `clusterctl describe cluster` present an overview of the cluster, its infrastructure,
+  control plane, and other related objects.
+
+The final iteration of this proposal will focus on:
+
+1. To define a mechanism for bubbling up/aggregating conditions up to hierarchy of cluster
+   objects
+
+
+### Non-Goals/Future Work
+
+Non-Goals
+
+- To pre-define every possible condition.
+- To remove  `Phase`, `ErrorMessage`, `ErrorReason` from Status in v0.3 .
+- To modify any Cluster API code in v0.3 to rely on conditions to determine actions or workflows.
+
+Future Work (v1alpha4)
+
+- To add core conditions in the Cluster API contract, eventually deprecating `Phase`, `ErrorMessage`,
+  `ErrorReason` from Status (TBD).
+- To modify Cluster API code to rely on conditions to determine actions or workflows.
+
+## Proposal
+
+Disclaimer: the goal of this first iteration is to to agree on a design for the condition types. 
+The condition types included in this document are only meant to validate the type design. There are
+no guarantees such examples will become real conditions during next iterations.
+
+### User Stories
+
+- Detail the things that people will be able to do if this proposal is implemented.
+- Include as much detail as possible so that people can understand the "how" of the system.
+- The goal here is to make this feel real for users without getting bogged down.
+
+#### Story 1
```

Comment:
I was expecting to see more user stories, including end user perspectives

Reply:
done!